src/client.js

```diff
@@ -412,44 +412,40 @@ Client.prototype.open = function(socketId, target) {
 };
 
 Client.prototype.sort = function(data) {
-	var self = this;
+	const order = data.order;
 
-	var type = data.type;
-	var order = data.order || [];
-	var sorted = [];
+	if (!_.isArray(order)) {
+		return;
+	}
 
-	switch (type) {
+	switch (data.type) {
 	case "networks":
-		order.forEach(i => {
-			var find = _.find(self.networks, {id: i});
-			if (find) {
-				sorted.push(find);
-			}
+		this.networks.sort((a, b) => {
+			return order.indexOf(a.id) > order.indexOf(b.id);
 		});
```

Comment:
Assuming about Node's (unstable) sorting algorithm, this is at best `O(n² * log n)`. Some people hang out in easily dozens of channels: 50 channels and that's 14k operations, 100 channels is 64k operations, etc.

I know this is not worse than it was before (I think), but do you think there is any way we can improve that? The user provides the order themselves, so in my understanding we shouldn't have to do a full sort, instead loop through the given order (which would be `O(n²)`, not much better, surely there is a better way...)

Also, when is this needed exactly? When a user reorders their channels/networks on the UI?